added demo page

Nikcodes10 opened the pull request:
## Issue that this pull request solves

 Closes: #195 

### Brief description of what is fixed or changed

Added Demo page
Home page now contains 2 buttons: Demo and Pro
Added Go back button to Navbar
Displays Logout button in Navbar only when user is logged-in
Demo page shows alert and hence become not-usable after 60 seconds.

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

![Screenshot (344)](https://user-images.githubusercontent.com/62133737/104356542-1725a180-5532-11eb-83b5-85ef6ac16c0e.png)
![Screenshot (345)](https://user-images.githubusercontent.com/62133737/104356554-1a209200-5532-11eb-8b2b-53de7a5d2855.png)
![Screenshot (346)](https://user-images.githubusercontent.com/62133737/104356568-1c82ec00-5532-11eb-8257-f400f6ff4674.png)


Any other information that is important to this pull request

I am a DSOC and SWOC participant

goliakshay357 commented on src/app/components/home/home.component.html:
```diff
@@ -40,7 +40,10 @@
-        <button class="btn-grad" routerLink="/login" >Try it out!</button>
+        <div class="row">
+          <button class="btn-grad" routerLink="/demo" >Demo</button>
+          <button class="btn-grad" routerLink="/login" >Pro</button>
```
Hey, 
There is nothing like pro in canvasboard. "TRY IT OUT" button will directly go to demo page (i.e demo component)

goliakshay357 commented:
Your build is failing. Please checkout in Details. 
Thanks

kingavatar commented:
Can you also please provide gif or mp4 preview of your UI changes. Thank you.

Nikcodes10 commented:
Ok

Nikcodes10 commented:
Removed demo button
Try it out button goes to demo page
Added Sign-in button to Navbar

![Screenshot (349)](https://user-images.githubusercontent.com/62133737/104609624-3dba1880-56a9-11eb-9a7c-e48e06505ac1.png)
![Screenshot (348)](https://user-images.githubusercontent.com/62133737/104609635-3f83dc00-56a9-11eb-9980-008e6625a906.png)

Why is the build failing?

Only reason I can see:

Warning: that 2mb budget exceeded

![Screenshot (347)](https://user-images.githubusercontent.com/62133737/104609812-70641100-56a9-11eb-9353-9e38b08cd554.png)

Do run locally and let me know if anything is wrong.

goliakshay357 commented:
Hello @Nikcodes10,
The build is failing at this stage (see attached scr shot)
![image](https://user-images.githubusercontent.com/32712438/104626299-bcb84c80-56bb-11eb-9626-a08087a1442b.png)

Just confirming, did you take the latest pull before making a PR?
Thanks

Kajol-Kumari commented:
Hey @Nikcodes10 do checkout [here](https://github.com/Canvasbird/canvasboard/pull/328/checks?check_run_id=1702724880#step:6:20) Seems like you forgot to import it in the test file of the `demoComponent` and that's making the test to fail. Check this once and it should be good to go!

Nikcodes10 commented:
@Kajol-Kumari I tried that. But it throws an error regarding the import of the Modules such as Location, HttpClient and Router. 

The problem is simple: Importing modules like AuthService, Router and Location into constructor of Demo component throws an error during tests. 

I don't know how to fix it.

I have fixed it for now by copying the navbar code and making it non-functional. In the ngOnInit function, I have commented what to do.

Kajol-Kumari commented:
Hey @Nikcodes10 I won't recommend copy-pasting the whole code instead of selector. Just make sure to follow [this](https://stackoverflow.com/questions/43937387/if-selector-is-an-angular-component-then-verify-that-it-is-part-of-this-mod) and your build should pass. Then this PR will be good to go, we can ignore the deepsource error if it doesn't get fixed after the build fix

Nikcodes10 commented:
Fixed everything. 
It was a simple problem of not importing the HttpClientTestingModule and the RouterTestingModule into the spec.ts files of demo and navbar components.

Kajol-Kumari reviewed (changes requested):
Hey @Nikcodes10 please do tye following-
1. Take the latest pull and resolve the conflicts. 
2. Remove unnecessary file changes like that if `debug.log` and `package-lock.json`

Nikcodes10 commented:
@Kajol-Kumari 
DWOC
